protocol/urls/urls.go

```diff
@@ -148,6 +161,46 @@ func GetYoutubePreviewData(link string) (previewData LinkPreviewData, err error)
 	return previewData, nil
 }
 
+func GetTwitterOembed(url string) (data TwitterOembedData, err error) {
+	oembedLink := fmt.Sprintf(TwitterOembedLink, url)
+	jsonBytes, err := GetURLContent(oembedLink)
+	if err != nil {
+		return data, fmt.Errorf("can't get bytes from twitter oembed response on %s link", oembedLink)
+	}
+
+	err = json.Unmarshal(jsonBytes, &data)
+	if err != nil {
+		return data, fmt.Errorf("can't unmarshall json")
```

Comment:
You might want to wrap the original error, something like:

```
return data, fmt.Errorf("can't unmarshall json: %w", err)
```
Similarly a few lines below

https://blog.golang.org/go1.13-errors#TOC_3.3.